Add documentation for MTA New York City Transit integration

OnFreund opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Add initial documentation for the MTA integration which provides real-time NYC subway arrival predictions using GTFS-RT data.


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/156846
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/8499
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

claude committed:
Add documentation for MTA New York City Transit integration

Add initial documentation for the MTA integration which provides
real-time NYC subway arrival predictions using GTFS-RT data.

c0ffeeca7 commented on source/_integrations/mta.markdown:
```diff
@@ -0,0 +1,71 @@
+ha_category:
+  - Transport
+ha_iot_class: Cloud Polling
+ha_release: '2025.12'
```
```suggestion
ha_release: 2026.1
```

c0ffeeca7 commented on source/_integrations/mta.markdown:
```diff
@@ -0,0 +1,71 @@
+## Configuration
+
+During setup, you will be asked to provide the following information:
+
```
```suggestion
```
the Configuration title is part of the include you added above.

c0ffeeca7 commented on source/_integrations/mta.markdown:
```diff
@@ -0,0 +1,71 @@
+
+The sensor includes the following attributes:
+
+- **stop_id**: The transit stop identifier.
```
```suggestion
- **Stop ID**: The transit stop identifier.
```
Are the UI strings really spelled with underscore?

c0ffeeca7 commented on source/_integrations/mta.markdown:
```diff
@@ -0,0 +1,71 @@
+The sensor includes the following attributes:
+
+- **stop_id**: The transit stop identifier.
+- **arrivals**: A list of upcoming train arrivals with details for each.
```
```suggestion
- **Arrivals**: A list of upcoming train arrivals with details for each.
```

c0ffeeca7 commented on source/_integrations/mta.markdown:
```diff
@@ -0,0 +1,71 @@
+
+Each item in the `arrivals` list contains:
+
+- **route**: The train route identifier.
```
```suggestion
- **Route**: The train route identifier.
```

c0ffeeca7 commented on source/_integrations/mta.markdown:
```diff
@@ -0,0 +1,71 @@
+Each item in the `arrivals` list contains:
+
+- **route**: The train route identifier.
+- **destination**: The final destination of the train.
```
```suggestion
- **Destination**: The final destination of the train.
```

c0ffeeca7 commented on source/_integrations/mta.markdown:
```diff
@@ -0,0 +1,71 @@
+
+- **route**: The train route identifier.
+- **destination**: The final destination of the train.
+- **minutes_until**: Minutes remaining until the train arrives.
```
```suggestion
- **Minutes until**: Minutes remaining until the train arrives.
```
Would minutes until arrival be an option?

c0ffeeca7 commented on source/_integrations/mta.markdown:
```diff
@@ -0,0 +1,71 @@
+- **route**: The train route identifier.
+- **destination**: The final destination of the train.
+- **minutes_until**: Minutes remaining until the train arrives.
+- **arrival_time**: The scheduled arrival time in ISO format.
```
```suggestion
- **Arrival time**: The scheduled arrival time in ISO format.
```
Would Scheduled arrival time be more clear, to differnetiate it from Minutes until arrival?

OnFreund committed:
Apply suggestions from code review

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

OnFreund committed:
Revise sensor entity description and attributes

Updated the description of the integration's sensor functionality and attributes.

OnFreund committed:
Apply suggestion from @c0ffeeca7

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

c0ffeeca7 commented on source/_integrations/mta.markdown:
```diff
@@ -0,0 +1,61 @@
+
+{% include common-tasks/define_custom_polling.md %}
+
+The default polling interval is 30 seconds.
```
```suggestion
The default {% term polling %} interval is 30 seconds.
```

c0ffeeca7 committed:
add glossary term reference

c0ffeeca7 reviewed (dismissed):
Thank you, @OnFreund :+1:

joostlek commented on source/_integrations/mta.markdown:
```diff
@@ -0,0 +1,61 @@
+ha_category:
+  - Transport
+ha_iot_class: Cloud Polling
+ha_release: 2026.1
```
```suggestion
ha_release: 2026.3
```

joostlek committed:
Update source/_integrations/mta.markdown
